### ISSUE TITLE
"org.klesun.deep_assoc_completion" -> "unable to get stub builder for file with file": composer.json

### ISSUE THREAD

**Issue by voku** [klesun/deep-assoc-completion#173]
I don't know why I saw this error, it's working as expected? Btw. thanks for the awesome plugin. :+1: 

```
java.lang.Throwable: unable to get stub builder for file with file = jar:///home/lmoelleken/.local/share/JetBrains/Toolbox/apps/PhpStorm/ch-0/203.6682.180.plugins/Symfony Support/lib/symfony-plugin-0.22.205.jar!/doctrine-meta/composer.json, file type = com.intellij.json.JsonFileType@3fb426b9, indexed file type = com.intellij.json.JsonFileType@3fb426b9, , stubBindingRoot = JsonFileImpl, languages = [JSON], fileTypes = [JSON], files = [JsonFileImpl], roots = []
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:159)
	at com.intellij.psi.stubs.StubProcessingHelperBase.handleNonPsiStubs(StubProcessingHelperBase.java:115)
	at com.intellij.psi.stubs.StubProcessingHelperBase.processStubsInFile(StubProcessingHelperBase.java:66)
	at com.intellij.psi.stubs.StubIndexImpl.lambda$processElements$2(StubIndexImpl.java:291)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:327)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:105)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:93)
	at com.jetbrains.php.PhpIndexImpl.getVariablesByName(PhpIndexImpl.java:477)
	at com.jetbrains.php.completion.PhpLookupElement.getNamedElement(PhpLookupElement.java:330)
	at com.jetbrains.php.completion.PhpLookupElement.getPhpType(PhpLookupElement.java:359)
	at com.jetbrains.php.completion.PhpCompletionContributor$PhpExpectedFunctionArgumentsCompletionProvider.collectLocalElementsByType(PhpCompletionContributor.java:2352)
	at com.jetbrains.php.completion.PhpCompletionContributor$PhpExpectedFunctionArgumentsCompletionProvider.fillExpectedFunctionCallsWithArgumentsSet(PhpCompletionContributor.java:2324)
	at com.jetbrains.php.completion.PhpCompletionContributor$PhpExpectedFunctionArgumentsCompletionProvider.addCompletions(PhpCompletionContributor.java:2306)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:32)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:155)
	at com.jetbrains.php.completion.PhpCompletionContributor.fillCompletionVariants(PhpCompletionContributor.java:454)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:77)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:154)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:146)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:142)
	at org.klesun.deep_assoc_completion.completion_providers.UsedStrValsPvdr.addCompletions(UsedStrValsPvdr.java:297)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:32)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:155)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:77)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:154)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:146)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:142)
	at com.intellij.codeInsight.template.impl.LiveTemplateCompletionContributor$1.addCompletions(LiveTemplateCompletionContributor.java:88)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:32)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:155)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:77)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:60)
	at com.intellij.codeInsight.completion.CompletionService.performCompletion(CompletionService.java:133)
	at com.intellij.codeInsight.completion.BaseCompletionService.performCompletion(BaseCompletionService.java:41)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.lambda$calculateItems$12(CompletionProgressIndicator.java:862)
	at com.intellij.util.indexing.FileBasedIndex.lambda$ignoreDumbMode$0(FileBasedIndex.java:179)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.util.indexing.FileBasedIndexEx.ignoreDumbMode(FileBasedIndexEx.java:543)
	at com.intellij.util.indexing.FileBasedIndex.ignoreDumbMode(FileBasedIndex.java:178)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.calculateItems(CompletionProgressIndicator.java:858)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.runContributors(CompletionProgressIndicator.java:846)
	at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lambda$startContributorThread$6(CodeCompletionHandlerBase.java:353)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$tryReadOrCancel$5(CompletionThreading.java:172)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1137)
	at com.intellij.codeInsight.completion.AsyncCompletion.tryReadOrCancel(CompletionThreading.java:170)
	at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lambda$startContributorThread$7(CodeCompletionHandlerBase.java:345)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$startThread$0(CompletionThreading.java:95)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:658)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:65)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$startThread$1(CompletionThreading.java:91)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:20)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:11)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.call(ApplicationImpl.java:270)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by klesun** [comment]
Thanks for the bug report!
An interesting thing, looks like some other plugin is failing when I call `runRemainingContributors()` when I get requested completion options. I guess I'll just add a try-catch there =-D.

**Comment by voku** [comment]
I don't know how or since when but some plugins has a built-in bug report in PhpStorm itself. I think more developers would report such bugs if you can submit it via one click.

![Bildschirmfoto von 2021-01-05 19-49-28](https://user-images.githubusercontent.com/264695/103686656-72362200-4f8f-11eb-92b0-6e4e1a58301a.png)

... ok wait :zany_face:  it's maybe not as good as it sounds: https://github.com/izhangzhihao/intellij-rainbow-brackets/issues?q=Stub+index+points+to+a+file+without+PS

**Comment by klesun** [comment]
Hopefully they won't need this button often =-D

**Comment by klesun** [comment]
Fixed in [2021.01.05.001](https://plugins.jetbrains.com/plugin/9927-deep-assoc-completion/versions/stable/107048)
